cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/DynamicDatasetCache.java

```diff
@@ -276,18 +341,20 @@ public boolean equals(Object o) {
 
       DatasetCacheKey that = (DatasetCacheKey) o;
 
-      return Objects.equal(this.name, that.name) &&
+      return Objects.equal(this.namespace, that.namespace) &&
+        Objects.equal(this.name, that.name) &&
         Objects.equal(this.arguments, that.arguments);
     }
 
     @Override
     public int hashCode() {
-      return Objects.hashCode(name, arguments);
+      return Objects.hashCode(namespace, name, arguments);
```

Comment:
same here... why not hash the access type? I know this is existing code. Perhaps can you find out who wrote this and ask whether there was a specific reason?